src/main/java/frc/robot/subsystems/swerve/gyro/gyropigeon2/GyroPigeon2.java

```diff
@@ -0,0 +1,42 @@
+package frc.robot.subsystems.swerve.gyro.gyropigeon2;
+
+import com.ctre.phoenix6.StatusSignal;
+import edu.wpi.first.math.geometry.Rotation2d;
+import frc.robot.constants.Phoenix6Constants;
+import frc.robot.constants.Ports;
+
+public class GyroPigeon2 {
+
+    private final GyroPigeon2Data gyroPigeon2Data;
+    private final GyroPigeon2Actions gyroPigeon2Actions;
+
+    public GyroPigeon2(){
+        GyroPigeon2ConfigObject gyroPigeon2ConfigObject = new GyroPigeon2ConfigObject(Ports.PIGEON_2_ID, Phoenix6Constants.CANIVORE_NAME);
+        this.gyroPigeon2Data = new GyroPigeon2Data(gyroPigeon2ConfigObject);
+        this.gyroPigeon2Actions = new GyroPigeon2Actions(gyroPigeon2ConfigObject.getGyro());
```

Comment:
the entire structure of this gyro is really weird, you make a new gyro configs object which creates the gyro motor and then you feed it into the data and actions. it just seems there is a nicer way to do this

Reply:
maybe create the pigeon in the class `GyroPigeon2`, and then make the `GyroPigeon2ConfigObject` get the motor in the constructor and the canivore name.
then you feed the status signals into `GyroPigeon2Data` and the motor into `GyroPigeon2Actions`

Reply:
also i dont like objects having "Object" in their name

Reply:
I can’t , the purpose of action is to do all the motor functions so that the main class pigeon gyro don’t need to deal with the motor. Same about data (which do also conversions it is not only status signals). 
In that way every class have a very specific and clear purpose and there is less room for error.

Reply:
Also with the current way you create in future more easily more gyros with same code if you some of them on robot.

Reply:
im not suggesting to move the motor functions to `GyroPigeon2`, im suggesting you move the creation of the motor to the `GyroPigeon2` constructor instead of the config class, and make the config class a status signals class.
that way you can feed only the status signals to the data and only the motor to the actions.
this makes less room for errors because the data only has the signals and not the motor anymore,the actions has the motor, and the `GyroPigeon2` doesn't have any of that.

Reply:
fine